### PR TITLE
issue #274 - Fix validation on scalar contained by array

### PR DESCRIPTION
**Pull request by mikaelcom**

### CODE REVIEW

File: tests/resources/generated/ValidApiArrayOfGuid.php [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    }
+    public static function validateStringForPatternConstraintFromSetString($values): string
+    {
+        $message = '';
+        $invalidValues = [];
+        foreach ($values as $arrayOfGuidStringItem) {
+            // validation for constraint: pattern([\da-fA-F]{8}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{12})
+            if (!preg_match('/[\\da-fA-F]{8}-[\\da-fA-F]{4}-[\\da-fA-F]{4}-[\\da-fA-F]{4}-[\\da-fA-F]{12}/', $arrayOfGuidStringItem)) {
+                $invalidValues[] = var_export($arrayOfGuidStringItem, true);
+            }
```
Comment (mikaelcom) [inline tests/resources/generated/ValidApiArrayOfGuid.php:80]: Would be better in the already generated method `validateStringForArrayConstraintsFromSetString`

File: tests/resources/generated/ValidApiArrayOfGuid.php [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    public function setString(?array $string = null): self
+    {
+        // validation for constraint: array
+        if ('' !== ($stringArrayErrorMessage = self::validateStringForArrayConstraintsFromSetString($string))) {
+            throw new InvalidArgumentException($stringArrayErrorMessage, __LINE__);
+        }
+        // validation for constraint: pattern([\da-fA-F]{8}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{12})
+        if ('' !== ($stringPatternErrorMessage = self::validateStringForPatternConstraintFromSetString($string))) {
+            throw new InvalidArgumentException($stringPatternErrorMessage, __LINE__);
+        }
```
Comment (mikaelcom) [inline tests/resources/generated/ValidApiArrayOfGuid.php:104]: Would be better in the already generated method `validateStringForArrayConstraintsFromSetString`

**Comment by mikaelcom** [comment]
## Coverage Report

|  Totals | Coverage |
| :-- | :-- |
| Statements: | 96.93% ( 4043 / 4171 ) |
| Methods: | 93.82% ( 713 / 760 ) |
| Lines: | 97.63% ( 3330 / 3411 ) |

### [StandWithUkraine](https://github.com/vshymanskyy/StandWithUkraine/blob/main/docs/README.md)

**Comment by mikaelcom** [comment]
## Coverage Report

|  Totals | Coverage |
| :-- | :-- |
| Statements: | 97% ( 4038 / 4163 ) |
| Methods: | 93.9% ( 708 / 754 ) |
| Lines: | 97.68% ( 3330 / 3409 ) |

### [StandWithUkraine](https://github.com/vshymanskyy/StandWithUkraine/blob/main/docs/README.md)

**Comment by mikaelcom** [comment]
## Coverage Report

|  Totals | Coverage |
| :-- | :-- |
| Statements: | 97% ( 4041 / 4166 ) |
| Methods: | 93.9% ( 708 / 754 ) |
| Lines: | 97.68% ( 3333 / 3412 ) |

### [StandWithUkraine](https://github.com/vshymanskyy/StandWithUkraine/blob/main/docs/README.md)

**Comment by mikaelcom** [comment]
## Coverage Report

|  Totals | Coverage |
| :-- | :-- |
| Statements: | 97% ( 4041 / 4166 ) |
| Methods: | 93.9% ( 708 / 754 ) |
| Lines: | 97.68% ( 3333 / 3412 ) |

### [StandWithUkraine](https://github.com/vshymanskyy/StandWithUkraine/blob/main/docs/README.md)

**Comment by mikaelcom** [comment]
## Coverage Report

|  Totals | Coverage |
| :-- | :-- |
| Statements: | 96.98% ( 4041 / 4167 ) |
| Methods: | 93.9% ( 708 / 754 ) |
| Lines: | 97.66% ( 3333 / 3413 ) |

### [StandWithUkraine](https://github.com/vshymanskyy/StandWithUkraine/blob/main/docs/README.md)

**Comment by mikaelcom** [comment]
## Coverage Report

|  Totals | Coverage |
| :-- | :-- |
| Statements: | 96.93% ( 4044 / 4172 ) |
| Methods: | 93.77% ( 708 / 755 ) |
| Lines: | 97.63% ( 3336 / 3417 ) |

### [StandWithUkraine](https://github.com/vshymanskyy/StandWithUkraine/blob/main/docs/README.md)

**Comment by mikaelcom** [comment]
## Coverage Report

|  Totals | Coverage |
| :-- | :-- |
| Statements: | 96.93% ( 4044 / 4172 ) |
| Methods: | 93.77% ( 708 / 755 ) |
| Lines: | 97.63% ( 3336 / 3417 ) |

### [StandWithUkraine](https://github.com/vshymanskyy/StandWithUkraine/blob/main/docs/README.md)

**Comment by mikaelcom** [comment]
## Coverage Report

|  Totals | Coverage |
| :-- | :-- |
| Statements: | 96.95% ( 4071 / 4199 ) |
| Methods: | 93.77% ( 708 / 755 ) |
| Lines: | 97.65% ( 3363 / 3444 ) |

### [StandWithUkraine](https://github.com/vshymanskyy/StandWithUkraine/blob/main/docs/README.md)

**Review by awaters-pa — APPROVED** [review]
This has tested well for me!

**Pull request merged**